lib/services/platform-service.ts

```diff
@@ -368,15 +368,56 @@ export class PlatformService implements IPlatformService {
 				this.$errors.failWithoutHelp("Unable to find built application. Try 'tns build %s'.", platform);
 			}
 
-			this.$fs.ensureDirectoryExists(path.dirname(targetPath)).wait();
+			function ext(file: string): string {
+				return path.extname(file).toLowerCase();
+			}
+			function targetExt(file: string): string {
+				let e = ext(file);
+				return (e === ".ipa" || e === ".app") ? e : ".";
+			}
+
+			switch(ext(packageFile) + "->" + targetExt(targetPath)) {
+				case ".ipa->.ipa":
+					this.$logger.trace(`Copying file '${packageFile}' to '${targetPath}'...`);
+					this.$fs.copyFile(packageFile, targetPath).wait();
+					this.$logger.info(`Copied file '${packageFile}' to '${targetPath}'.`);
+					break;
+
+				case ".app->.app":
+					this.$logger.trace(`Copying directory '${packageFile}' to '${targetPath}'...`);
+					shell.cp("-Rf", packageFile + "/*", targetPath);
```

Comment:
And I will probably remove the "f" flag. I've played a little with `xcodebuild` and it throws if you specify output path that already exists so we should probably stick to that behavior as you initially suggested.